File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    result = []
    
    arr = []
    time = 0
    cnt = 0
    
    while len(progresses) > 0:
        # 작업이 완료되었다면
        if (progresses[0] + time * speeds[0]) >= 100:
            progress = progresses.pop(0)
            speed = speeds.pop(0)
            cnt += 1
        else:
            # 완성된 기능이 있다면 배포
            if cnt > 0:
                result.append(cnt)
                cnt = 0
            
            time += 1
    
    result.append(cnt)
    
    return result

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    let queue = [];
    
    for (let i = 0; i < progresses.length; i++) {
        queue.push(Math.ceil((100 - progresses[i]) / speeds[i]));
    }
    
    let index = 1;
    let max = queue[0];
    let count = 1;
    
    while (index < queue.length) {
        if (queue[index] > max) {
            answer.push(count);
            max = queue[index];
            count = 1;
        } else {
            count++;
        }
        
        index++;
    }
    
    answer.push(count);
    
    return answer;
}